src/components/member/tabs/contract-tab/agreement-item.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react'
+import { Table } from 'semantic-ui-react'
+import { Button, ButtonsGroup } from '../../../../../shared/hedvig-ui/button'
+import { Card } from '../../../../../shared/hedvig-ui/card'
+import { Spacing } from '../../../../../shared/hedvig-ui/spacing'
+import { Agreement, Contract } from '../../../../api/generated/graphql'
+import { ButtonSpacing, FlexWrapper } from './contract-item'
+import { InsuranceCertificate } from './InsuranceCertificate'
+import { InsuranceMandate } from './InsuranceMandate'
+import { SwedishApartmentAgreementKeyInfo } from './swedish-apartment-agreement-key-info'
+import { ToDate } from './ToDateComponent'
+import { FromDate } from './FromDateComponent'
+import { WithShowNotification } from '../../../../store/actions/notificationsActions'
+
+export const AgreementItemComponent: React.FunctionComponent<{
+  agreement: Agreement
+  contract: Contract
+} & WithShowNotification> = ({ agreement, contract, showNotification }) => {
+  const getAgreementKeyInfo = () => {
+    if (agreement.__typename.toString() === 'SwedishApartment') {
```

Comment:
If this is the best way to check union types: For our own sanity we should probably add a helper function to make these checks. 
```suggestion
    if (isOfType(agreement, 'SwedishApartment')) {
```
Or if typescript supports it:
```suggestion
    if (agreement.isOfType('SwedishApartment') {
```

Reply:
I think `agreement.__typename === 'SwedishApartment'` should be sufficient for typescript to make out which type `agreement` is. That _should_ also let typescript know what properties are available from the union. At least we're doing this (but in a different context) in the web onboarding: https://github.com/HedvigInsurance/web-onboarding/blob/283af70263d98059bee745f3abf58629bd5802cd/src/pages/OfferNew/Checkout/index.tsx#L304-L313